Include/httpClient/config.h

```diff
@@ -16,7 +16,8 @@
 #define HC_DATAMODEL_LP64 3 // int is 32 bit; long and pointer are 64 bits (64 bit unix)
 // see http://www.unix.org/version2/whatsnew/lp64_wp.html for detailed descriptions
 
-#if defined(_WIN32)
+#if defined(HC_PLATFORM) && defined(HC_DATAMODEL)
```

Comment:
maybe move the check for defining a datamodel inside the if and output a helpful error for that?
something like: "if manually setting HC_PLATFORM also set HC_DATAMODEL"?